saleor/graphql/page/types.py

```diff
@@ -108,3 +111,13 @@ def resolve_has_pages(root: models.PageType, info, **kwargs):
             .load(root.pk)
             .then(lambda pages: bool(pages))
         )
+
+    @staticmethod
+    def __resolve_references(roots: List["PageType"], info, **_kwargs):
+        ids = [
+            int(from_global_id_or_error(root.id, PageType, raise_error=True)[1])
+            for root in roots
+        ]
+        qs = models.PageType.objects.filter(id__in=ids)
```

Comment:
Pages visibility depends on user's permission, see the `resolve_pages` resolver -it uses `visible_for_user` method. Could we reuse the resolver here? If needed I would adjust the resolver to accept an optional list of IDs. Ideally those resolver function should be the only places where we can easily maintain this user-specific visibility logic.

Reply:
Nevermind, I was thinking about Page model and not PageType. This resolver is fine.